src/pykeen/models/unimodal/boxe_kg.py

```diff
@@ -0,0 +1,196 @@
+                       widths_p1 * torch.abs(points - centres) - (0.5 * widths) * (widths_p1 - 1 / widths_p1))
+    return dist
+
+
+class BoxEKGInteraction(Interaction):
+    relation_shape = ('d', 'd', 's', 'd', 'd', 's')  # Boxes are 2xd (size) each, x 2 sets of boxes: head and tail
+    entity_shape = ('d', 'd')   # Base position and bump
+
+    def __init__(self, tanh_map=True, norm_order=2):
+        super().__init__()
+        self.tanh_map = tanh_map  # Map the tanh map
+        self.norm_order = norm_order
+
+    def forward(self, h, r, t):
+        rh_base, rh_delta, rh_size, rt_base, rt_delta, rt_size = r
+        h_pos, h_bump = h
+        t_pos, t_bump = t
+        # First, compute the boxes
+        rh_low, rh_high = compute_box(rh_base, rh_delta, rh_size)
+        rt_low, rt_high = compute_box(rt_base, rt_delta, rt_size)
+        # Second, compute bumped entity representations
+        # Normalization
+        points_h = (h_pos + t_bump)
+        points_t = (t_pos + h_bump)
+        # Third, optionally apply the tanh transformation
+        if self.tanh_map:
+            rh_low = torch.tanh(rh_low)
+            rh_high = torch.tanh(rh_high)
+            rt_low = torch.tanh(rt_low)
+            rt_high = torch.tanh(rt_high)
+
+            points_h = torch.tanh(points_h)
+            points_t = torch.tanh(points_t)
+        # Fourth, compute the dist function output
+        dist_h = dist(points_h, rh_low, rh_high)
+        dist_t = dist(points_t, rt_low, rt_high)
+        # Fifth, compute the norm
+        score_h = dist_h.norm(p=self.norm_order, dim=-1)
+        score_t = dist_t.norm(p=self.norm_order, dim=-1)
+        total_score = score_h + score_t
+        #NSSA = NSSALoss(margin=5, adversarial_temperature=0.0, reduction='sum')
+        #if total_score.shape[0] == 512:  # Positive facts hard code
+            #print("I'm getting the correct loss here!")
+            #print(NSSA.forward(pos_scores=-total_score, neg_scores=-100*total_score,
+            #                  neg_weights=torch.ones_like(total_score)))
+            #print("That's just a test")
+        return - total_score    # Because this is inverted in NSSALoss (higher is better)
+
+
+class BoxEKG(ERModel):
+    r"""An implementation of BoxE
+
+    ---
+    citation:
+        author: Abboud
+        year: 2020
+        link: https://arxiv.org/abs/2007.06267
+    """
+
+    #: The default strategy for optimizing the model's hyper-parameters
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        embedding_dim=DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE,
+        p=dict(type=int, low=1, high=2),
+    )
+
+    def __init__(
+        self,
+        *,
+        embedding_dim: int = 200,
+        tanh_map: bool = True,
+        norm_order: int = 2,
+        entity_initializer: Hint[Initializer] = uniform_,
+        entity_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_initializer: Hint[Initializer] = uniform_,    # Has to be scaled as well
```

Comment:
there's an alternative function for uniform + norm at `pykeen.nn.init.uniform_norm_`